web/src/components/userContext.tsx

```diff
@@ -0,0 +1,39 @@
+import {
+  ENDPOINT_CONFIGS,
+  GetCurrentUserRequest,
+  GetCurrentUserResponse,
+} from '@codersquare/shared';
+import { useQuery } from '@tanstack/react-query';
+import { ReactNode, createContext, useContext } from 'react';
+
+import { callEndpoint } from '../fetch';
+import { isLoggedIn } from '../fetch/auth';
+
+type UserContext = {
+  currentUser?: GetCurrentUserResponse;
+  refreshCurrentUser?: any;
```

Comment:
nit: type this as a function, maybe something like `refreshCurrentUser?: () => void`.